src/solid/WindowVirtualizer.tsx

```diff
@@ -116,11 +127,15 @@ export const WindowVirtualizer = <T,>(
     itemSize,
     overscan,
     undefined,
-    undefined,
+    cache,
     !itemSize
   );
   const resizer = createWindowResizer(store, horizontal);
   const scroller = createWindowScroller(store, horizontal);
+  // The elements length and cached items length are different just after element is added/removed.
+  if (props.data.length !== store.$getItemsLength()) {
+    store.$update(ACTION_ITEMS_LENGTH_CHANGE, [props.data.length, props.shift]);
+  }
```

Comment:
This will not work, and implementation already exists:
```suggestion
```

https://github.com/inokawa/virtua/blob/4ff43ff0061bd5f6c48bae0888631fa35b25887a/src/solid/WindowVirtualizer.tsx#L178-L187